cpp/src/io/json/nested_json.hpp

```diff
@@ -199,7 +200,21 @@ struct json_column {
                   uint32_t child_count)
   {
     // If, thus far, the column's type couldn't be inferred, we infer it to the given type
-    if (type == json_col_t::Unknown) { type = row_type; }
+    if (type == json_col_t::Unknown) {
+      type = row_type;
+    }
+    // If, at some point within a column, we encounter a nested type (list or struct),
+    // we change that columns type to that respective nested type and invalidate all previous rows
```

Comment:
Should we at least warn about this case? It seems easy to lose data that way

Reply:
We don't have any warning mechanisms in libcudf (AFAIK). There's a plan to add logging similar (or same) to what RMM has.